lib/ionic-project-transformator.ts

```diff
@@ -0,0 +1,416 @@
+///<reference path=".d.ts"/>
+"use strict";
+
+import * as path from "path";
+import * as shelljs from "shelljs";
+import * as xmlMapping from "xml-mapping";
+import Future = require("fibers/future");
+
+export class IonicProjectTransformator implements IIonicProjectTransformator {
+	private static IONIC_PROJECT_BACKUP_FOLDER_NAME = "Ionic_Backup";
+	private static CONFIG_XML_FILE_NAME = "config.xml";
+	private static ANDROID_PLATFORM_NAME = "android";
+	private static IOS_PLATFORM_NAME = "ios";
+	private static WINDOWS_PHONE_PLATFORM_NAME = "wp8";
+	private static ANDROID_XML_FOLDER_NAME = "xml";
+	private static SUPPORTED_SCREEN_SIZES = ["ldpi", "mdpi", "hdpi", "xhdpi", "xxhdpi", "xxxhdpi"];
+	private static SUPPORTED_LAYOUT_SIZES = ["small", "normal", "large", "xlarge"];
+	private static RESOURCE_TYPES = ["drawable", "mipmap"];
+	private static XML_RESOURCE_TYPES = ["layout"];
+
+	/// The first orientation is empty string because there can be folder for both orientations (e.g. drawable-hdpi).
+	private static SCREEN_ORIENTATIONS = ["", "land", "port"];
+	private static WINDOWS_PHONE_SUPPORTED_SPLASH_SCREEN_FORMAT = ".jpg";
+	private static WINDOWS_PHONE_IONIC_SPLASH_SCREEN_FORMAT = ".png";
+	private supportedScreensFolderNames: string[];
+	private _project: Project.IProject;
+	private _ionicResourcesDirectory: string;
+	private _ionicConfigXml: ConfigXmlFile.IConfigXmlFile;
+
+	constructor(private $fs: IFileSystem,
+		private $projectConstants: Project.IConstants,
+		private $logger: ILogger,
+		private $injector: IInjector) {
+		this.supportedScreensFolderNames = this.createSupportedScreensFoldernames();
+	}
+
+	private get project(): Project.IProject {
```

Comment:
can you name this `$project`, so when we look at the code, we'll easily understand that it's from the injector